assets/js/datagovsg-search.js

```diff
@@ -122,6 +137,41 @@ function remove(array, elements) {
   });
 }
 
+function displaySearchFilterDropdown(fields, startingField) {
+  var fieldFilterDesktop = document.getElementById('field-filter-desktop');
+  var fieldFilterMobile = document.getElementById('field-filter-mobile');
+
+  for (let field of fields) {
+    // Creating the select element for mobile view
+    var option = document.createElement("option");
+    option.value = field;
+    option.text = field;
+    if (field === startingField) option.selected = true
+    fieldFilterMobile.appendChild(option);
+
+    // Creating the a tags for desktop view
+    var a_element = document.createElement("a");
+    a_element.id = field;
+    a_element.classList.add("bp-dropdown-item", "py-0");
+    a_element.onclick = function () {
+      return function () {
+        var filterDropdownDesktop = document.getElementById('field-selector-desktop');
+        filterDropdownDesktop.value = field;
+      };
+    }();
+    if (field === startingField) {
+      var filterDropdownDesktop = document.getElementById('field-selector-desktop');
+      filterDropdownDesktop.value = field;
+    }
+
+    fieldFilterDesktop.appendChild(a_element);
+    var p_element = document.createElement("p");
+    p_element.innerHTML = field.replace("_", " ");
```

Comment:
Hmm this replacement doesn't seem to be happening:

![Screenshot 2023-10-26 at 14 19 31](https://github.com/isomerpages/isomerpages-template/assets/27919917/efdf5256-08f8-4085-88c5-c8645e99a72b)

Reply:
This is actually a partial replacement, which only helps the dropdown options! The actual field needs to be exact because it's being passed as a query param used in the actual search. Converting between the prettified form and the actual value to send would either need a hidden field or something else - I opted not to do this because it causes additional complications